docs/fml-spec.md

```diff
@@ -0,0 +1,576 @@
+    case .screenshot:
+        item.thumbnailData = tab.screenshot?.pngData()
+    case .none:
+        item.thumbnailData = nil
+}
+```
+
+Note that enumeration variants in Swift are in `mixedCamelCase`; in Kotlin they are in `SCREAMING_SNAKE_CASE`.
+
+## Feature defaults
+
+Already, in this simple example we have three variables. We may want to vary the configuration of the feature when the user is not involved in an experiment.
+
+This might be because we have run an experiment and learned that a change of configuration should be made more permanent.
+
+The `defaults` list is a list of zero or more "default blocks", used to patch the default values of the variables in a feature.
+
+The `value` property defines the patch which is overlaid on top of the existing variable values.
+
+```yaml
+features:
+  spotlight-search:
+    variables:
+      enabled:
+        description: If `false`, the app will not record anything with Spotlight.
+        type: Bool
+        default: false
+      max-age-in-days:
+        description: The number of days a single piece of content is indexed for.
+        type: Double
+        default: 28.0
+      item-thumbnail:
+        description: "The icon that appears in the Spotlight search results.
+          Note that changing this does not change already indexed content."
+        type: ThumbnailType
+        default: letter
+    defaults:
+      - value: { item-thumbnail: screenshot, max-age-in-days: 56.0 }
+```
+
+The `defaults` list of `value`s set the default value of `item-thumbnail` to `screenshot`, and `max-age-in-days` to `56.0`.
+
+Adding new feature blocks to the `defaults` list patches the existing defaults further:
+
+```yaml
+    defaults:
+      - value: { item-thumbnail: screenshot, max-age-in-days: 56.0 }
+      - value: { max-age-in-days: 64.0 }
+```
+
+In this case, the default value for `item-thumbnail` is now `screenshot`, and `max-age-in-days` is `64.0`.
+
+### Feature defaults and channels
+
+For release engineering reasons, or marketing reasons, we might want to ensure that the feature isn't enabled by default for the release population of users.
+
+However, for quality reasons, we might want a different configuration for our nightly audience.
+
+This is achieved by using channels.
+
+```yaml
+channels:
+  - nightly
+  - beta
+  - release
+features:
+  spotlight-search:
+    variables:
+      enabled:
+        description: "…"
+        type: Bool
+        default: false
+    defaults:
+      - channel: nightly
+        value: { enabled: true }
+```
+
+The `channel` property specifies which build flavour the default applies. In this example, the `nightly` version of the app set the `enabled` variable to `true`; all other versions of the app have `enabled` as `false`.
+
+In this way, app developers can use the feature manifest as a replacement for other more adhoc feature flag solutions.
+
+### Feature defaults and runtime targeting
```

Comment:
I think you should leave anything that is unimplemented out and put it in a separate proposal to avoid confusion about the current state of what's being implemented.

Also, I think it would be line up with a use case that needs to use it (search or pocket come to mind, particularly around geos) to ensure it's meeting their needs.

Reply:
I've filtered those out into a separate file.